File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        lo, hi = 0, len(numbers) - 1
        while lo < hi:
            ss = numbers[lo] + numbers[hi]
            if ss == target:
                return [lo + 1, hi + 1]
            elif ss < target:
                lo+=1
            else:
                hi-=1
        return res

File: Data Structures & Algorithms/4sum/submission-1.java
class Solution {
    public List<List<Integer>> fourSum(int[] nums, int target) {
       Set<List<Integer>> res = new HashSet<>();
       int L = nums.length;
       Arrays.sort(nums);
       for(int i=0; i < L; i++) {
        for(int j=i+1; j < L; j++) {
            int le=j+1, ri = L - 1;
            while(le < ri) {
                long ss = ((long)nums[i]+(long)nums[j]+(long)nums[le]+(long)nums[ri]);
                if (ss == target) {
                    res.add(Arrays.asList(nums[i],nums[j],nums[le],nums[ri]));
                    le++;
                    ri--;
                } else if(ss < target) {
                    le++;
                } else {
                    ri--;
                }
            }
        }
       }
       return new ArrayList<>(res); 
    }
}